features/features.go

```diff
@@ -0,0 +1,75 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package features
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+// SnapdFeature is a named feature that may be on or off.
+type SnapdFeature int
+
+const (
+	// Layouts controls availability of snap layouts.
+	Layouts SnapdFeature = iota
+	// ParallelInstances controls availability installing a snap multiple times.
+	ParallelInstances
+	// Hotplug controls availability of dynamically creating slots based on system hardware.
+	Hotplug
+	// SnapdSnap controls possibility of installing the snapd snap.
+	SnapdSnap
+	// PerUserMountNamespace controls the persistence of per-user mount namespaces.
+	PerUserMountNamespace
+	// PerformanceMeasurements controls the collection of performance data.
+	PerformanceMeasurements
+)
+
+// String returns the name of a snapd feature.
+func (f SnapdFeature) String() string {
+	// The constants here must be synchronized with cmd/libsnap-confine-private/feature.c
+	switch f {
+	case Layouts:
+		return "layouts"
+	case ParallelInstances:
+		return "parallel-instances"
+	case Hotplug:
+		return "hotplug"
+	case SnapdSnap:
+		return "snapd-snap"
+	case PerUserMountNamespace:
+		return "per-user-mount-namespace"
+	case PerformanceMeasurements:
+		return "performance-measurements"
+	}
+	panic(fmt.Sprintf("unknown feature flag code %d", f))
+}
+
+// IsEnabled checks if a given snapd feature is enabled.
+func (f SnapdFeature) IsEnabled() bool {
+	fi, err := os.Stat(filepath.Join(dirs.FeaturesDir, f.String()))
```

Comment:
Why not just `osutil.FileExists()`?

Reply:
Ah, I thought we had something like that but couldn't find it quickly. I'll use it :)